src/utils.sh

```diff
@@ -8,51 +8,96 @@ VERSION_TAG="v0.0.0-rc0"
 # create flag:variable_name dictionary
 declare -A flag_to_variable_dict
 
-add_option_args() {
+######################################
+# add_option_args
+#
+#    declare a valid command-line argument(s)
+#
+# ARGUMENTS:
+#    OPTION_NAME    The name of the argument (e.g "VERBOSE")
+#    a list of one or more argument switches (e.g "-v", "--verbose")
+# OUTPUTS:
+#    Write output to stdout
+# RETURN:
+#
+######################################
+
+function add_option_args() {
     if [[ $# > 1 ]];
     then
-        local option_flag=$1; shift
-        local var_name=$1; shift
-        flag_to_variable_dict[$option_flag]=$var_name
+        local option_name=$1; shift
+        while [ $# -ne 0 ];
+        do
+            flag_to_variable_dict[$1]=$option_name
+            shift
+        done
     fi
 }
 
-# generic command line parser
+######################################
+# clear_option_args
+#
+#    clears the list of valid command-line arguments
+#
+# ARGUMENTS:
+#
+# OUTPUTS:
+#
+# RETURN:
+#
+######################################
+
+function clear_option_args() {
+    flag_to_variable_dict=()
+}
+
 function cmd_parser() {
-    # create flag:variable_name dictionary
+    # create flag:variable_name dictionary and initialize to empty string
     declare -A parsed_cmd
+    for key in ${!flag_to_variable_dict[*]};
+    do
+        parsed_cmd[${flag_to_variable_dict[$key]}]=""
+    done
 
-    for arg in "$@"
+    while [ $# -ne 0 ];
     do
-        if [[ $arg == -* ]];
+        if [[ $1 == -* ]];
         then
-            # iterate over all the keys in dictionary
-            local valid_arg="false"
-            for k in ${!flag_to_variable_dict[*]};
+            local valid_argument=false
+            # for each key in the dictionary
+            for key in ${!flag_to_variable_dict[*]};
             do
-                # if arg==key, then we store into flag_to_val_dict 
-                if [ "$arg" == "$k" ];
-                then 
-                    # set parsed_cmd, which is varname:value
-                    parsed_cmd[${flag_to_variable_dict[$k]}]=$2
-                    valid_arg="true"
+                if [ "$1" == "$key" ]
+                then
+                    valid_argument=true
+                    if [[ $# == 1 || $2 == -* ]];
+                    then
+                        parsed_cmd[${flag_to_variable_dict[$key]}]=true
+                    else
+                        parsed_cmd[${flag_to_variable_dict[$key]}]=$2
+                        shift
+                    fi
                     break
                 fi
             done
 
             # found an unknown argument
-            if [[ "$valid_arg" == "false" ]];
+            if [[ "$valid_argument" == "false" ]];
             then
                 parsed_cmd=()
```

Comment:
Let user know when the arguments are invalid?

Reply:
I would like to leave this for a later iteration. I think it is a large enough scope to require its own PBI.